.github/workflows/build-release-binaries.yml

```diff
@@ -0,0 +1,104 @@
+name: Build and Release Binaries
+
+on:
+  workflow_run:
+    workflows: ["Publish Release"]
+    types:
+      - completed
+
+jobs:
+  build-binaries:
+    # Only run if the triggering workflow succeeded and it was triggered by a tag
+    if: >
+      github.event.workflow_run.conclusion == 'success' &&
+      startsWith(github.event.workflow_run.head_branch, 'refs/tags/v')
+    runs-on: ubuntu-24.04
+    name: Build Cross-Platform Binaries
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.23.4'
+
+    - name: Get app/version.Version from the code
+      if: github.ref_type == 'branch'
+      run: |
+        VERSION=$(grep 'var version' app/version/version.go | cut -d'"' -f2)
+        echo "APP_VERSION=$VERSION" >> $GITHUB_ENV
+        echo "RELEASE_VERSION=$VERSION" >> $GITHUB_ENV
+
+    - name: Set version from git tag
+      if: github.ref_type == 'tag'
+      run: |
+        echo "APP_VERSION=${{ github.ref_name }}" >> $GITHUB_ENV
+        echo "RELEASE_VERSION=${{ github.ref_name }}" >> $GITHUB_ENV
+        echo "GO_BUILD_FLAG=-ldflags=-X github.com/obolnetwork/charon/app/version.version=${{ github.ref_name }}" >> $GITHUB_ENV
+
+    - name: Set build flags for branch
+      if: github.ref_type == 'branch'
+      run: |
+        echo "GO_BUILD_FLAG=-ldflags=-X github.com/obolnetwork/charon/app/version.version=${{ env.APP_VERSION }}" >> $GITHUB_ENV
+
+    - name: Install cross-compilation dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y build-essential gcc-aarch64-linux-gnu
+
+    - name: Build x86_64 binary
+      run: |
+        mkdir -p dist
+        docker run --rm --platform linux/amd64 \
+          -v $(pwd):/workspace -w /workspace \
+          golang:1.23.4-bookworm \
```

Comment:
Hope we can make this `golang:1.23.4-bookworm` matching our globally set golang version..
We typically bump it in one place.